include/swift/AST/AutoDiff.h

```diff
@@ -77,79 +78,193 @@ class AutoDiffParameter {
   }
 };
 
+/// SIL-level automatic differentiation indices. Consists of a source index,
+/// i.e. index of the dependent result to differentiate from, and parameter
+/// indices, i.e. index of an independent parameter to differentiate with
+/// respect to.
+struct SILReverseAutoDiffIndices {
+  unsigned source;
+  llvm::BitVector parameters;
+  
+  /*implicit*/ SILReverseAutoDiffIndices(unsigned source,
+                                         llvm::BitVector parameters)
+    : source(source), parameters(parameters) {}
+  
+  /*implicit*/ SILReverseAutoDiffIndices(unsigned source,
+                                         ArrayRef<unsigned> parameters)
```

Comment:
Would it make sense for clients to pass in a bitvector, instead of array<unsigned>?   If array<unsigned> is the right input, please move this method out of line.

Reply:
This is passed from the parser. We don't know the number of indices ahead of time, everything needs to be appended to a `SmallVector` first. This happens at various places in the parser, so I had this c'tor for convenience.